adhocracy4/emails/tasks.py

```diff
@@ -1,16 +1,48 @@
 import importlib
+import logging
+from typing import Optional
 
+from celery import Task
 from celery import shared_task
 from django.apps import apps
 
+log = logging.getLogger(__name__)
 
-@shared_task
+
+@shared_task(bind=True)
 def send_async(
-    email_module_name, email_class_name, app_label, model_name, object_pk, args, kwargs
+    self: Task,
+    email_module_name: str,
+    email_class_name: str,
+    app_label: str,
+    model_name: str,
+    object_pk: Optional[int] = None,
+    args: Optional[list] = None,
+    kwargs: Optional[dict] = None,
+    data: Optional[dict] = None,
```

Comment:
Since we refactoring, I would use the kwargs for the data (I mentioned it in our DM), otherwise, the data after the kwargs while both are dicts is redundant. Name it **kwargs or **data, as you prefer. The kwargs are passed to the context I saw, so that should be working.
Then put back the asterisks that denote what type those args and kwargs/data are, and skip the hinting for those, except we are sure what type args and kwargs expected to be inside the tuple/dict, but we aren't. [PEP484](https://peps.python.org/pep-0484/#arbitrary-argument-lists-and-default-argument-values)
 In emails/base.py the dispatch() and get_context() need refactoring too, to work with object optionally. If it stays as it is, code remains confusing, as we pass data which is a dict, and then is called object in EmailBase.